File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1.sort()
        nums2.sort()
        result = []

        first_point = second_point = 0

        while first_point < len(nums1) and second_point < len(nums2):
            if nums1[first_point] < nums2[second_point]:
                first_point += 1
            elif nums1[first_point] > nums2[second_point]:
                second_point += 1
            else:
                result.append(nums1[first_point])
                first_point += 1
                second_point += 1
        return result                 
        

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.dart
class Solution {
  List<int> intersect(List<int> nums1, List<int> nums2) {  
     Map<int, int> freqMap = {};
     List<int> result = [];
   
    for(int n in nums1) {
      freqMap[n] = (freqMap[n] ?? 0) + 1;
    }
  
    for(int n in nums2) {
      if(freqMap.containsKey(n) && (freqMap[n] ?? 0) > 0) {
         result.add(n);
         freqMap[n] = (freqMap[n] ?? 0) - 1;
      }
    }
    
    return result;  
    
  }
}